src/lib/context.py

```diff
@@ -65,6 +68,28 @@ def exception(self, *args):
         self.error(*args)
         traceback.print_exc()
 
+    def t_error(self, *args):
+        """
+        Prints error with throttling limit per message. Limit per message set in LOG_THROTTLING_LIMIT_PER_CALLER
+        Use for potentially frequent log.
+        :param args:
+        :return:
+        """
+        if args and self.__check_if_message_exceeded_limit(args[-1]):
```

Comment:
Just checking, but isn't this approach not very optimal? If an error were to be repeatedly called, but each time with slightly different arguments, it would keep counting it separately.

Reply:
As to the issue of passing the caller id to be printed, you could just pass all but the last argument from t_error to error

Reply:
You're right, that with this approach messages with different arguments will be logged. I've decided to count occurrences per message on purpose. 
We get rid of log spam of already logged errors and exceptions and we keep logging messages with different arguments. These messages might be crucial for troubleshooting and I believe that troubleshooting is the main purpose of logging.